core/domain/collection_domain.py

```diff
@@ -257,11 +259,50 @@ def create_default_node(cls, exploration_id):
         return cls(exploration_id, [], [])
 
 
+class CollectionSkill(object):
+    """Domain object describing a skill in the collection.
+
+    The skill contains the skill id, the human readable name, and the list of
+    question IDs associated to the skill.
+    """
+
+    def __init__(self, skill_id, name, question_ids):
+        """Constructs a new CollectionSkill object.
+
+        Args:
+        - skill_id (int): the skill ID
+        - name (string): the displayed name of the skill
+        - question_ids (list of strings): list of the question IDs associated
+                                          to the skill.
+        """
+        self.skill_id = skill_id
+        self.name = name
+        self.question_ids = question_ids
+
+    def to_dict(self):
+        return {
+            'skill_id': self.skill_id,
+            'name': self.name,
+            'question_ids': self.question_ids
+        }
+
+    @classmethod
+    def from_dict(cls, skill_dict):
+        return cls(
+            copy.deepcopy(skill_dict['skill_id']),
+            copy.deepcopy(skill_dict['name']),
+            copy.deepcopy(skill_dict['question_ids'])
+        )
+
+    # TODO(wxy): validation, and other skill functions go here
+
+
 class Collection(object):
     """Domain object for an Oppia collection."""
 
     def __init__(self, collection_id, title, category, objective,
-                 language_code, tags, schema_version, nodes, version,
+                 language_code, tags, schema_version, nodes, skills,
+                 skills_id_count, version,
```

Comment:
For one data point, I prefer ``skill_id_count`` :) but I don't have a strong preference